### PR TITLE
[REVIEW] Add binary and comparison ops to dataframe, tests.

### LINKED ISSUE #900: [FEA] Implement division on DataFrame objects
```python
import cudf
df = cudf.DataFrame({'x': [1, 2, 3]})
df / df
```

```python-traceback
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-3-28e3f97d0440> in <module>()
----> 1 df / df

TypeError: unsupported operand type(s) for /: 'DataFrame' and 'DataFrame'
```

I believe that there is already another issue for this, but I thought I'd repost this particular operation in case it's easier to implement quickly.   This is useful for groupby aggregations with dask dataframe.

### PR DESCRIPTION
**Pull request by thomcom**
Pretty straight forward.

- binaryops
- comparisonops
- tests

Fixes #900

### CODE REVIEW

File: python/cudf/dataframe/dataframe.py [hunk 1]
```diff
@@ -408,6 +409,127 @@ def __repr__(self):
...
+                else:
+                    result[col] = Series(cudautils.full(self.shape[0],
+                                         np.dtype('float64').type(np.nan),
+                                         'float64'), nan_as_null=False)
+            for col in self._cols:
+                if col not in other._cols:
+                    result[col] = Series(cudautils.full(self.shape[0],
+                                         np.dtype('float64').type(np.nan),
+                                         'float64'), nan_as_null=False)
+        elif isinstance(other, Series):
```
Comment (kkraus14) [inline python/cudf/dataframe/dataframe.py]: I imagine Index is the same as here, but we should have an `isinstance(other, Index)`block as well.

File: python/cudf/dataframe/dataframe.py [hunk 2]
```diff
@@ -408,6 +409,127 @@ def __repr__(self):
...
+                    "Series to DataFrame arithmetic not supported "
+                    "until strings can be used as indices. Try converting your"
+                    " Series into a DataFrame first.")
+        elif isinstance(other, numbers.Number):
+            for col in self._cols:
+                result[col] = getattr(self._cols[col], internal_fn)(
+                        other,
+                        fn,
+                )
+        else:
```
Comment (kkraus14) [inline python/cudf/dataframe/dataframe.py]: What would trigger this else clause?

File: python/cudf/dataframe/dataframe.py (hunk 1 above)
Comment (kkraus14) [reply python/cudf/dataframe/dataframe.py]: We can do this in a followup PR when we enable ops on Indexes in general though.

File: python/cudf/dataframe/dataframe.py (hunk 2 above)
Comment (thomcom) [reply python/cudf/dataframe/dataframe.py]: I'm going to go with "nothing we support". I don't know why that is in there, a restructuring of the earlier clauses.

Comment (thomcom) [reply python/cudf/dataframe/dataframe.py]: See my fix.

File: python/cudf/tests/test_dataframe.py [hunk 3]
```diff
@@ -1292,9 +1304,13 @@ def test_reductions(pdf, gdf, accessor, func):
...
-def test_binops(pdf, gdf, left, right, binop):
-    d = binop(left(pdf), right(pdf))
-    g = binop(left(gdf), right(gdf))
+def test_binops_series(pdf, gdf, binop):
+    pdf = pdf + 1.0
+    gdf = gdf + 1.0
+    d = binop(pdf.x, pdf.y)
+    g = binop(gdf.x, gdf.y)
+    print(d)
+    print(g)
```
Comment (mrocklin) [inline python/cudf/tests/test_dataframe.py]: Print statements

File: python/cudf/tests/test_dataframe.py [hunk 4]
```diff
@@ -1292,9 +1304,13 @@ def test_reductions(pdf, gdf, accessor, func):
...
 ])
-def test_binops(pdf, gdf, left, right, binop):
-    d = binop(left(pdf), right(pdf))
-    g = binop(left(gdf), right(gdf))
+def test_binops_series(pdf, gdf, binop):
+    pdf = pdf + 1.0
+    gdf = gdf + 1.0
+    d = binop(pdf.x, pdf.y)
+    g = binop(gdf.x, gdf.y)
+    print(d)
```
Comment (kkraus14) [inline python/cudf/tests/test_dataframe.py]:
```suggestion
```

File: python/cudf/tests/test_dataframe.py (hunk 3 above)
Comment (kkraus14) [reply python/cudf/tests/test_dataframe.py]:
```suggestion
```

**Pull request merged**